### PR TITLE
Feature/post status validation

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

**Comment by Schparky** [comment]
Is it possible/acceptable to skip Accepted? Use case: carriers Commits and delivers without waiting for the receiver Accepting. Can't control this because it's outside the scope of the app (real-world user behavior). Question is, do we require the receiver to first accept an item that has been delivered, or can they just say it has been delivered, skipping Accepted?

File: application/domain/domain.go [hunk 1]
```diff
@@ -197,8 +198,16 @@ func RollbarMiddleware(next buffalo.Handler) buffalo.Handler {
 
 // Error log error and send to Rollbar
 func Error(c buffalo.Context, msg string, extras map[string]interface{}) {
+
+	// Avoid panics running tests when c doesn't have the necessary nested methods
+	cType := fmt.Sprintf("%T", c)
+	if cType == "models.EmptyContext" {
+		return
+	}
+
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/domain/domain.go]: I tried the c.(type) thing, but the type is `models.EmptyContext` which would need to be imported, which then creates  circular dependencies.

File: application/models/post.go [hunk 2]
```diff
@@ -96,16 +96,84 @@ func (p *Post) Validate(tx *pop.Connection) (*validate.Errors, error) {
...
+		PostStatusCommitted: {PostStatusOpen, PostStatusAccepted, PostStatusRemoved},
+		PostStatusAccepted:  {PostStatusOpen, PostStatusReceived, PostStatusRemoved},
+		PostStatusReceived:  {PostStatusCompleted},
+		PostStatusCompleted: {},
+		PostStatusRemoved:   {},
+	}
+
+	goodStatuses, ok := okTransitions[oldPost.Status]
+	if !ok {
+		return
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/models/post.go]: Done

File: application/models/post_test.go [hunk 3]
```diff
@@ -142,6 +142,475 @@ func (ms *ModelSuite) TestPost_Validate() {
...
+		FirstName: "Existing",
+		LastName:  "User",
+		Nickname:  "Existing User ",
+		Uuid:      domain.GetUuid(),
+	}
+
+	if err := ms.DB.Create(&user); err != nil {
+		t.Errorf("could not create test user %v ... %v", user.Email, err)
+		t.FailNow()
+	}
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/models/post_test.go]: Done

File: application/models/post_test.go [hunk 4]
```diff
@@ -142,6 +142,475 @@ func (ms *ModelSuite) TestPost_Validate() {
...
+		Url:        nulls.String{},
+		AuthType:   AuthTypeSaml,
+		AuthConfig: "{}",
+		Uuid:       domain.GetUuid(),
+	}
+	err := ms.DB.Create(org)
+	if err != nil {
+		t.Errorf("Failed to create organization for test, error: %s", err)
+		t.FailNow()
+	}
```
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/models/post_test.go]: Done

**Pull request merged**